core/src/main/java/org/apache/calcite/interpreter/Bindables.java

```diff
@@ -545,6 +557,58 @@ public RelNode convert(RelNode rel) {
     }
   }
 
+  /**
+   * Rule to convert an {@link org.apache.calcite.rel.logical.LogicalIntersect}
+   * to a {@link BindableIntersect}.
+   */
+  public static class BindableIntersectRule extends ConverterRule {
+
+    /**
+     * Creates a BindableIntersectRule.
+     *
+     * @param relBuilderFactory Builder for relational expressions
+     */
+    public BindableIntersectRule(RelBuilderFactory relBuilderFactory) {
+      super(LogicalIntersect.class, (Predicate<RelNode>) r -> true,
+          Convention.NONE, BindableConvention.INSTANCE, relBuilderFactory,
+          "BindableIntersectRule");
+    }
+
+    public RelNode convert(RelNode rel) {
+      final LogicalIntersect intersect = (LogicalIntersect) rel;
+      final BindableConvention out = BindableConvention.INSTANCE;
+      final RelTraitSet traitSet = intersect.getTraitSet().replace(out);
+      return new BindableIntersect(rel.getCluster(), traitSet,
+          convertList(intersect.getInputs(), out), intersect.all);
+    }
+  }
+
+  /**
+   * Rule to convert an {@link org.apache.calcite.rel.logical.LogicalMinus}
+   * to a {@link BindableMinus}.
+   */
+  public static class BindableMinusRule extends ConverterRule {
```

Comment:
Can we have one `BindableXXXRule` and one `BindableXXX` to unify all the set operators ?

Reply:
I think we need to reconginze SqlKind in SetOp, if we run a sql ,  we would like to see operators: BindableIntersect、BindableMinus、Bindable rather than BindableSetOp.

Reply:
But at least the rule can be unified.

Reply:
yeah, will update it.